TOLTECS: Add keymapper support

aunnoman1 opened the pull request:
<!---
Thank you for contributing to ScummVM. Please read the following carefully before submitting your Pull Request.

Make sure your individual commits follow the guidelines found in the ScummVM Wiki: https://wiki.scummvm.org/index.php?title=Commit_Guidelines. If they're not please edit them before submitting the Pull Request.

Proper documentation must also be included for common code and changes impacting user facing elements.

Commits and Pull Requests should use the following template:

```
SUBSYSTEM: Short (50 chars or less) summary of changes

More detailed explanatory text, if necessary.  Wrap it to about 72
characters or so.  In some contexts, the first line is treated as the
subject of an email and the rest of the text as the body.  The blank
line separating the summary from the body is critical (unless you omit
the body entirely); tools like rebase can get confused if you run the
two together.

Write your commit message in the present tense: "Fix bug" and not "Fixed
bug."  This convention matches up with commit messages generated by
commands like git merge and git revert.

Further paragraphs come after blank lines.

- Bullet points are okay, too

- Typically a hyphen or asterisk is used for the bullet, preceded by a
 single space, with blank lines in between, but conventions vary here

- Use a hanging indent
```
--->

sev- commented on engines/toltecs/menu.cpp:
```diff
@@ -85,10 +85,17 @@ int MenuSystem::run(MenuID menuId) {
 		_returnToGame = true;
 
+	Common::Keymapper *keymapper = _vm->getEventManager()->getKeymapper();
+	keymapper->getKeymap("toltecs-default")->setEnabled(false);
```
```suggestion
    // Now we turn off the keymapper so it does not interfere with full text input
	keymapper->getKeymap("toltecs-default")->setEnabled(false);
```

sev- commented on engines/toltecs/movie.cpp:
```diff
@@ -271,12 +271,16 @@ bool MoviePlayer::handleInput() {
-			if (event.kbd.keycode == Common::KEYCODE_ESCAPE)
+		case Common::EVENT_CUSTOM_ENGINE_ACTION_START:
+			switch (event.customType) {
+			case (kActionSkipMovie):
```
why?

```suggestion
			case kActionSkipMovie:
```

and it other places

sev- commented on engines/toltecs/toltecs.h:
```diff
@@ -38,6 +38,9 @@
 
 #include "toltecs/console.h"
 
+#include "backends/keymapper/action.h"
```
Are these needed here? We try to minimize include dependencies.

sev- reviewed (changes requested):
Good shape, I have only couple of minor notes

OMGPizzaGuy commented on engines/toltecs/metaengine.cpp:
```diff
@@ -199,6 +205,59 @@ SaveStateDescriptor ToltecsMetaEngine::querySaveMetaInfos(const char *target, in
+	act = new Action(kStandardActionLeftClick, _("Move / Select"));
+	act->setLeftClickEvent();
+	act->addDefaultInputMapping("MOUSE_LEFT");
+	engineKeyMap->addAction(act);
```
I recommend a few of the defaults for JOY buttons as well from the now overridden MetaEngine::initKeymaps.

OMGPizzaGuy commented on engines/toltecs/toltecs.h:
```diff
@@ -83,6 +83,16 @@ enum SysString {
+	kActionSkipMovie,
+	kActionMenuOpen,
+	kActionSkipRide,
+	kActionNone,
```
I recommend kActionNone at the top of the action enum as it seems an appropriate default value

aunnoman1 commented:
Thank you for the suggestions @OMGPizzaGuy . I've implemented them.

ALso fixed an issue which made joystick binds not work in game menu by readjusting where the keymaps are enabled and disabled
